src/Symfony/Component/Translation/Translator.php

```diff
@@ -50,9 +53,14 @@ class Translator implements TranslatorInterface, TranslatorBagInterface
     private $resources = array();
 
     /**
-     * @var MessageSelector
+     * @var MessageFormatterInterface[] An array of formatters
      */
-    private $selector;
+    private $formatters = array();
+
+    /**
+     * @var string
+     */
+    private $defaultFormatter = TransMessageFormatter::NAME;
```

Comment:
A better idea would be to take the name of the first formatter in the constructor